tests/testthat/test-separate_longer.R

```diff
@@ -0,0 +1,491 @@
+  df <- pl$DataFrame(
+    id = 1:3,
+    x = c("a,b", "c,d", "e"),
+    y = c("abcd", "efg", "hi"),
+    z = c(1L, 2L, 3L)
+  )
+
+  expect_equal(df |> separate_longer_delim_polars(z, delim = ","), df)
+  expect_equal(df |> separate_longer_position_polars(z, width = 2), df)
+
+  expect_equal(
+    df |> separate_longer_delim_polars(c(x, z), delim = ","),
+    df |> separate_longer_delim_polars(x, delim = ",")
+  )
+  expect_equal(
+    df |> separate_longer_position_polars(c(y, z), width = 2),
+    df |> separate_longer_position_polars(y, width = 2)
+  )
+})
+
+# Error tests
+test_that("errors on non-polars data", {
+  df <- data.frame(id = 1:2, x = c("a,b", "c,d"))
+
+  expect_snapshot(
+    df |> separate_longer_delim_polars(x, delim = ","),
+    error = TRUE
+  )
+  expect_snapshot(
+    df |> separate_longer_position_polars(x, width = 2),
+    error = TRUE
+  )
+})
+
+test_that("errors on non-existent column", {
+  df <- pl$DataFrame(
+    id = 1:2,
+    x = c("a,b", "c,d")
+  )
+
+  expect_snapshot(
+    df |> separate_longer_delim_polars(nonexistent, delim = ","),
+    error = TRUE
+  )
+
+  expect_snapshot(
+    df |> separate_longer_position_polars(nonexistent, width = 2),
+    error = TRUE
+  )
+})
+
+test_that("errors when cols is missing", {
+  df <- pl$DataFrame(
+    id = 1:2,
+    x = c("a,b", "c,d")
+  )
+
+  expect_snapshot(
+    df |> separate_longer_delim_polars(delim = ","),
+    error = TRUE
+  )
+
+  expect_snapshot(
+    df |> separate_longer_position_polars(width = 2),
+    error = TRUE
+  )
+})
+
+test_that("errors when delim is missing", {
+  df <- pl$DataFrame(
+    id = 1:2,
+    x = c("a,b", "c,d")
+  )
+
+  expect_snapshot(
+    df |> separate_longer_delim_polars(x),
+    error = TRUE
+  )
+})
+
+test_that("errors when width is missing", {
+  df <- pl$DataFrame(
+    id = 1:2,
+    x = c("a,b", "c,d")
+  )
+
+  expect_snapshot(
+    df |> separate_longer_position_polars(x),
+    error = TRUE
+  )
+})
+
+test_that("errors when width is invalid", {
+  df <- pl$DataFrame(
+    id = 1:2,
+    x = c("ab", "cd")
+  )
+
+  expect_snapshot(
+    df |> separate_longer_position_polars(x, width = 0),
+    error = TRUE
+  )
+
+  expect_snapshot(
+    df |> separate_longer_position_polars(x, width = 1.5),
+    error = TRUE
+  )
+})
+
+test_that("errors when ... is not empty", {
+  df <- pl$DataFrame(
+    id = 1:2,
+    x = c("a,b", "c,d")
+  )
+
+  expect_snapshot(
+    df |> separate_longer_delim_polars(x, delim = ",", extra = TRUE),
+    error = TRUE
+  )
+
+  expect_snapshot(
+    df |> separate_longer_position_polars(x, width = 2, extra = TRUE),
+    error = TRUE
+  )
+})
+
+# Error when n to m (n, m >= 2, n != m) - incompatible lengths
+test_that("separate_longer_delim_polars errors on incompatible lengths", {
+  df <- pl$DataFrame(
+    id = 1:2,
+    x = c("a,b,c", "d,e"),
+    y = c("1,2", "3,4,5,6")
+  )
+
+  expect_snapshot(
+    df |> separate_longer_delim_polars(c(x, y), delim = ","),
+    error = TRUE
+  )
```

Comment:
Below this, we can test that tidypolars and tidyr both error in this scenario with 
```r
expect_equal_or_both_error(
  df |> separate_longer_delim_polars(c(x, y), delim = ","), 
  df |> as.data.frame() |> separate_longer_delim(c(x, y), delim = ",")
)
````
That way, if tidyr's behavior changes for some reason in the future, we'll know that tidypolars' behavior will also need to be updated.

Same for the test below.